test/jdk/jdk/nio/zipfs/NewFileSystemTests.java

```diff
@@ -170,6 +177,92 @@ public void testNewFileSystemPathNullMap() {
                 FileSystems.newFileSystem(Path.of("basic.jar"), nullMap));
     }
 
+    /**
+     * Validate that without {@code "create" = true}, a ZIP file system cannot be
+     * opened if the underlying file is missing, but even with this set, a ZIP
+     * file system cannot be opened for conflicting or invalid access modes.
+     */
+    @DataProvider(name = "badEnvMap")
+    protected Object[][] badEnvMap() {
+        return new Object[][]{
+                {Map.of(), NoSuchFileException.class},
+                {Map.of("accessMode", "readOnly"), NoSuchFileException.class},
+                {Map.of("accessMode", "readWrite"), NoSuchFileException.class},
+                {Map.of("create", true, "accessMode", "readOnly"), IllegalArgumentException.class},
+                {Map.of("create", true, "accessMode", "badValue"), IllegalArgumentException.class},
+        };
+    }
+    @Test(dataProvider = "badEnvMap")
+    public void badArgumentsFailure(Map<String, String> env, Class<? extends Throwable> exception) throws IOException {
+        assertThrows(exception, () -> FileSystems.newFileSystem(Path.of("no_such.zip"), env));
+    }
+
+    /**
+     * Validate that multi-release JARs can be opened read-write if no release
+     * version is specified.
+     */
+    @Test
+    public void multiReleaseJarReadWriteSuccess() throws IOException {
+        // Multi-release JARs, when opened with a specified version are inherently read-only.
+        Path multiReleaseJar = createMultiReleaseJar();
+        try (FileSystem fs = FileSystems.newFileSystem(multiReleaseJar, Map.of("accessMode", "readWrite"))) {
+            assertFalse(fs.isReadOnly());
+            if (!"Default version".equals(Files.readString(fs.getPath("file.txt"), UTF_8))) {
+                fail("unexpected file content");
+            }
+        }
+    }
+
+    /**
+     * Validate that when the underlying file is read-only, it cannot be opened in
+     * read-write mode.
+     */
+    @Test
+    public void readOnlyZipFileFailure() throws IOException {
+        // Underlying file is read-only.
+        Path readOnlyZip = Utils.createJarFile("read_only.zip", Map.of("file.txt", "Hello World"));
+        readOnlyZip.toFile().setReadOnly();
+        assertThrows(IOException.class,
+                () -> FileSystems.newFileSystem(readOnlyZip, Map.of("accessMode", "readWrite")));
+    }
+
+    /**
+     * Validate that multi-release JAR is opened read-only by default if a release
+     * version is specified.
+     */
+    @Test
+    public void multiReleaseJarDefaultReadOnly() throws IOException {
+        Path multiReleaseJar = createMultiReleaseJar();
+        try (FileSystem fs = FileSystems.newFileSystem(multiReleaseJar, Map.of("releaseVersion", "1"))) {
+            assertTrue(fs.isReadOnly());
+            if (!"First version".equals(Files.readString(fs.getPath("file.txt"), UTF_8))) {
+                fail("unexpected file content");
```

Comment:
Nit, for this and the other new `fail` statements, it might be good to include the unexpected file content in the fail message. If at all the test fails for whatever reason, that additional detail usually help to quickly debug the failures. Or maybe just replace the `if` followed by a `fail` with an `assertEquals()` call.